### PR TITLE
Host off switch

### PR DESCRIPTION
**Pull request by DavidVorick**
This PR is not complete, just wanted to post it for review. Still needs testing, API support, and siac support.

Commit messages are pretty strong.

### CODE REVIEW

**Comment by DavidVorick** [comment]
```
--- FAIL: TestIntegrationWalletTransactionGETid (0.01s)

	wallet_test.go:245: provided encryption key is incorrect
```

This needs to be inspected. It's nondeterministic.

**Comment by DavidVorick** [comment]
```
--- FAIL: TestReceiveConsensusSetUpdate (0.48s)

	update_test.go:83: hostdb should have a host after getting a host announcement transcation
```
I've seen this nondeterministic error a few times as well.

**Comment by DavidVorick** [comment]
```
--- FAIL: TestIntegrationRenewing (7.87s)

	host_test.go:114: 0

	host_test.go:115: uploading has failed
```

Another failure. Hmmm, I hope there's a common thread to all of these.

**Comment by lukechampine** [comment]
Why not treat `AcceptingContracts` like any other HostSettings field? i.e., to toggle it on or off you would go through the existing `/host [POST]` call, with `/host?acceptingcontracts=false`.
It's not as user-friendly as a separate toggle, but that shouldn't matter since UIs can easily make it a separate call if so desired.

File: modules/host/upload.go [hunk 1]
```diff
@@ -252,11 +252,20 @@ func (h *Host) managedNegotiateContract(conn net.Conn, filesize uint64, merkleRo
 // scheme, file contracts should not initially hold any data.
 func (h *Host) managedRPCUpload(conn net.Conn) error {
 	// Check that the host has grabbed an address from the wallet.
-	h.mu.RLock()
+	h.mu.Lock()
```
Comment (lukechampine) [inline modules/host/upload.go]: why do you need a write lock here?

**Comment by DavidVorick** [comment]
derp. Designed this in the reverse order, added the switch first and later decided it should be a setting. Didn't even think through the fact that we can already adjust the settings at will.

We can drop the extra methods and api calls, this change has a lot more code than is needed.

**Pull request closed without merging**